packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_CoarseOperator_def.hpp

```diff
@@ -675,61 +688,61 @@ namespace FROSch {
             for (UN j=1; j<GatheringMaps_.size(); j++) {
                 {
 #ifdef FROSCH_COARSEOPERATOR_DETAIL_TIMERS
-                    FROSCH_TIMER_START_LEVELID(coarseSolveExportersTime,"Build Exporters");
+                    FROSCH_DETAILTIMER_START_LEVELID(coarseSolveExportersTime,"Build Exporters");
 #endif
                     CoarseSolveExporters_[j] = ExportFactory<LO,GO,NO>::Build(GatheringMaps_[j-1],GatheringMaps_[j]);
                     CoarseSolveExporters_[j]->setDistributorParameters(gatheringCommunicationList); // Set the parameter list for the communication of the exporter
                 }
 #ifdef FROSCH_COARSEOPERATOR_EXPORT_AND_IMPORT
                 {
 #ifdef FROSCH_COARSEOPERATOR_DETAIL_TIMERS
-                    FROSCH_TIMER_START_LEVELID(coarseSolveImportersTime,"Build Importers");
+                    FROSCH_DETAILTIMER_START_LEVELID(coarseSolveImportersTime,"Build Importers");
 #endif
                     CoarseSolveImporters_[j] = ImportFactory<LO,GO,NO>::Build(GatheringMaps_[j],GatheringMaps_[j-1]);
                     CoarseSolveImporters_[j]->setDistributorParameters(gatheringCommunicationList); // Set the parameter list for the communication of the exporter
                 }
 #endif
             }
-        } else if(!DistributionList_->get("Type","linear").compare("ZoltanDual")){
-          //ZoltanDual provides a partition of the coarse problem with Zoltan2 inlcuding the
-          //build of a Repeated map suited for the next level
-          //GatheringSteps to communicate Matrix
-          int gatheringSteps = DistributionList_->get("GatheringSteps",1);
-          GatheringMaps_.resize(gatheringSteps);
-          CoarseSolveExporters_.resize(gatheringSteps);
-
-          double gatheringFactor = pow(double(this->MpiComm_->getSize())/double(NumProcsCoarseSolve_),1.0/double(gatheringSteps));
-          LO numProcsGatheringStep = this->MpiComm_->getSize();
-          GO numGlobalIndices = CoarseMap_->getMaxAllGlobalIndex();
-          GO numMyRows;
-          numMyRows = 0;
-
-          if (this->MpiComm_->getRank()%(this->MpiComm_->getSize()/NumProcsCoarseSolve_) == 0 && this->MpiComm_->getRank()/(this->MpiComm_->getSize()/NumProcsCoarseSolve_) < NumProcsCoarseSolve_) {
-            if (this->MpiComm_->getRank()==0) {
-                numMyRows = numGlobalIndices - (numGlobalIndices/NumProcsCoarseSolve_)*(NumProcsCoarseSolve_-1);
-            } else {
-                numMyRows = numGlobalIndices/NumProcsCoarseSolve_;
+        } else if (!DistributionList_->get("Type","linear").compare("ZoltanDual")) {
```

Comment:
Should this be guarded with ` #ifdef HAVE_SHYLU_DDFROSCH_ZOLTAN2` ?

Reply:
See my comment above. However, you are right. Can you add this to your list, @roeverf?